cms/static/js/spec/views/video_transcripts_spec.js

```diff
@@ -189,6 +228,161 @@ define(
                     verifyTranscriptActions($transcriptEl.find('.transcript-actions'), languageCode);
                 });
             });
+
+            it('can upload transcript', function() {
+                var languageCode = 'en',
+                    newLanguageCode = 'ar',
+                    requests = AjaxHelpers.requests(this),
+                    $transcriptEl = videoTranscriptsView.$el.find('.show-video-transcript-content[data-language-code="' + languageCode + '"]'); // eslint-disable-line max-len
+
+                    // Verify correct transcript title is set.
+                expect($transcriptEl.find('.transcript-title').html()).toEqual(
+                    'Video client title n_' + languageCode + '.' + TRANSCRIPT_DOWNLOAD_FILE_FORMAT
+                );
+
+                // Select a language
+                $transcriptEl.find('.transcript-language-menu').val(newLanguageCode);
+
+                $transcriptEl.find('.upload-transcript-button').click();
+
+                // Add transcript to upload queue and send POST request to upload transcript.
+                $transcriptEl.find('.upload-transcript-input').fileupload('add', {files: [createFakeTranscriptFile()]});
+
+                // Verify if POST request received for image upload
+                AjaxHelpers.expectRequest(
+                    requests,
+                    'POST',
+                    TRANSCRIPT_UPLOAD_URL
+                );
+
+                // Send successful upload response
+                AjaxHelpers.respondWithJson(requests, {});
+
+                // Verify correct transcript title is set.
+                expect($transcriptEl.find('.transcript-title').html()).toEqual(
+                    'Video client title n_' + newLanguageCode + '.' + TRANSCRIPT_DOWNLOAD_FILE_FORMAT
+                );
+
+                verifyMessage($transcriptEl, 'uploaded');
+
+                // Verify transcript actions are rendered correctly.
+                verifyTranscriptActions($transcriptEl.find('.transcript-actions'), newLanguageCode);
```

Comment:
What about the transcript language dropdown's state changes. We should also assert those.